packages/rich-text-structure/src/create.js

```diff
@@ -0,0 +1,326 @@
+		const { start, end, text, formats } = createRecord( node, range, settings );
+		const length = accumulator.text.length;
+
+		if ( range ) {
+			if ( start !== undefined ) {
+				accumulator.start = length + start;
+			} else if (
+				node.parentNode === range.startContainer &&
+				node === range.startContainer.childNodes[ range.startOffset ]
+			) {
+				accumulator.start = length;
+			}
+
+			if ( end !== undefined ) {
+				accumulator.end = length + end;
+			} else if (
+				node.parentNode === range.endContainer &&
+				node === range.endContainer.childNodes[ range.endOffset - 1 ]
+			) {
+				accumulator.end = length + text.length;
+			}
+		}
+
+		accumulator.formats = accumulator.formats.concat( formats );
+		accumulator.text += text;
+
+		if ( index !== maxIndex ) {
+			accumulator.formats = accumulator.formats.concat( [ , , ] );
+			accumulator.text += '\n\n';
+		}
+
+		return accumulator;
+	}, emptyRecord );
+}
+
+/**
+ * Creates a rich text value object from a DOM element.
+ *
+ * @param {HTMLElement} element      Element to create value object from.
+ * @param {string}      multilineTag Multiline tag.
+ * @param {Object}      settings     Settings passed to `createRecord`.
+ *
+ * @return {Object} A rich text value object.
+ */
+export function createValue( element, multilineTag, settings ) {
+	return create( element, null, multilineTag, settings );
+}
+
+/**
+ * Creates rich text value and selection objects from a DOM element and range.
+ *
+ * @param {HTMLElement} element                  Element to create value object
+ *                                               from.
+ * @param {Range}       range                    Range to create selection object
+ *                                               from.
+ * @param {Object}      settings                 Settings object.
+ * @param {Function}    settings.removeNodeMatch Function to declare whether the
+ *                                               given node should be removed.
+ * @param {Function}    settings.unwrapNodeMatch Function to declare whether the
+ *                                               given node should be unwrapped.
+ * @param {Function}    settings.filterString    Function to filter the given
+ *                                               string.
+ * @param {Function}    settings.removeAttribute Match Wether to remove an attribute
+ *                                               based on the name.
+ *
+ * @return {Object} A rich text record.
+ */
+function createRecord( element, range, settings = {} ) {
+	const emptyRecord = {
+		formats: [],
+		text: '',
+	};
+
+	if ( ! element || ! element.hasChildNodes() ) {
+		return emptyRecord;
+	}
+
+	const {
+		removeNodeMatch = () => false,
+		unwrapNodeMatch = () => false,
+		filterString = ( string ) => string,
+		removeAttributeMatch,
+	} = settings;
+
+	// Remove any line breaks in text nodes. They are not content, but used to
+	// format the HTML. Line breaks in HTML are stored as BR elements.
```

Comment:
It seems like we cannot really assume `\n`
https://www.w3.org/TR/html5/syntax.html#writing-text

maybe this instead?
```js
string.replace( /[\r\n]/, '' )
```

Reply:
Thanks! I just saw that this won't replace all line breaks either. Updating and adding a test.